test/e2e/storage/flexvolume.go

```diff
@@ -72,56 +72,60 @@ func testFlexVolume(driver string, cs clientset.Interface, config framework.Volu
 // installFlex installs the driver found at filePath on the node, and restarts
 // kubelet if 'restart' is true. If node is nil, installs on the master, and restarts
 // controller-manager if 'restart' is true.
-func installFlex(c clientset.Interface, node *v1.Node, vendor, driver, filePath string, restart bool) {
+func installFlex(c clientset.Interface, node *v1.Node, vendor, driver, filePath string) {
 	flexDir := getFlexDir(c, node, vendor, driver)
 	flexFile := path.Join(flexDir, driver)
 
 	host := ""
+	var err error
 	if node != nil {
-		host = framework.GetNodeExternalIP(node)
+		host, err = framework.GetNodeExternalIP(node)
+		if err != nil {
+			host, err = framework.GetNodeInternalIP(node)
+		}
 	} else {
-		host = net.JoinHostPort(framework.GetMasterHost(), sshPort)
+		masterHostWithPort := framework.GetMasterHost()
+		hostName := getHostFromHostPort(masterHostWithPort)
+		host = net.JoinHostPort(hostName, sshPort)
+	}
+
+	if host == "" {
```

Comment:
Do we need this check?

Reply:
I think yes for printing proper error when we can't determine external or internal ip for some reason. Before this change  `GetNodeExternalIP` used to fail on it.

Reply:
Sorry I wasn't clear. I meant that
```go
if host == "" {
    	framework.ExpectNoError(err)
}
```
is equivalent to only
```
framework.ExpectNoError(err)
```

because there's an error only if host == "" in the implementations of `GetNode*ternalIP()`

Reply:
fixed